src/common/xattr.h

```diff
@@ -19,12 +19,14 @@
 extern "C" {
 #endif
 
+#if defined(__linux__)
```

Comment:
i am fine with this change, but it's always good to have `ENOATTR` defined, right? or we need to check if `ENODATA` is defined also here? to avoid cyclic definition? as we have
```
#define ENODATA ENOATTR
```
in `compat.h`.

[OT] if `xattr.c` is the only consumer of `ENOATTR`, maybe we can move this definition into `xattr.c` in another commit.